pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -358,7 +360,19 @@ func checkDaemonSystemdService() error {
 	logging.Debug("Checking crc daemon systemd service")
 
 	// the daemon should not be running at the end of setup, as it must be restarted on upgrades
-	return checkSystemdUnit(daemonUnitName, daemonUnitContent(), false)
+	alreadyRunningErr := checkSystemdUnit(daemonUnitName, daemonUnitContent(), false)
+	if alreadyRunningErr == nil {
+		return nil
+	}
+	if !errors.Is(alreadyRunningErr, unitShouldNotBeRunningErr(daemonUnitName)) {
+		return alreadyRunningErr
+	}
```

Comment:
got confused with the name `alreadyRunningErr` it could be either `unitShouldNotBeRunningErr` or the generic "file content mismatch" error, and in case of the letter its an "`alreadyRunningErr` plus file content different" maybe just name it as `err` but now that i understand it its okay either way or a comment would be helpful!

Reply:
I can rename it, I gave the `alreadyRunningErr` name before adding the `unitStatusErr` type, and did not adjust the name.
I did not name it `err` because when we fail to get the version from the daemon, I prefer to return `alreadyRunningErr` rather than the error which was returned from the `GetVersion` call:
```
        // daemon is running, check its version
	version, err := daemonclient.GetVersionFromDaemonAPI()
	if err != nil {
		return alreadyRunningErr
	}
```

Reply:
Updated.